File: ClassicalAlgos/BFS_algorithm/BFS.py
# BFS python implementation to find path from the source vertex in a given graph
from collections import defaultdict 
class Graph: 
    # Graph initialization
	def __init__(self): 
		self.graph = defaultdict(list) 
	def addEdge(self,u,v): 
		self.graph[u].append(v)
	def BFS(self, s):
		visited = [0]*(len(self.graph)) # initializing all nodes as not visited
		queue = [] 
		queue.append(s) 
		visited[s] = 1 # initializing source as visited
		while queue: # traversing through all nodes reachable
			s = queue.pop(0) 
			print (s, end = "  ")
			for i in self.graph[s]: 
				if visited[i] == 0: 
					queue.append(i) 
					visited[i] = 1
# sample graph to run
g = Graph() 
g.addEdge(0, 1); 
g.addEdge(0, 2); 
g.addEdge(1, 2); 
g.addEdge(2, 3); 
g.BFS(2) 
 

File: searchingAlgo/breadthFirstSearch/BFS.java
// BFS java implementation to find path from the source vertex in a given graph
import java.io.*; 
import java.util.*; 
class Graph 
{ 
	private int V; 
	private LinkedList<Integer> adj[]; 
	Graph(int v) // initialize graph
	{ 
		V = v; 
		adj = new LinkedList[v]; 
		for (int i=0; i<v; ++i) 
			adj[i] = new LinkedList(); 
	}  
	void addEdge(int v,int w) // add edges
	{ 
		adj[v].add(w); 
	} 
	void BFS(int s) // algorithm
	{ 
		boolean visited[] = new boolean[V]; 
		LinkedList<Integer> queue = new LinkedList<Integer>(); 
		visited[s]=true; 
		queue.add(s); 

		while (queue.size() > 0) 
		{  
			s = queue.poll(); 
			System.out.print(s+" "); 
			Iterator<Integer> i = adj[s].listIterator(); 
			while (i.hasNext()) 
			{ 
				int n = i.next(); 
				if (!visited[n]) 
				{ 
					visited[n] = true; 
					queue.add(n); 
				} 
			} 
		} 
	} 
	public static void main(String args[]) // sample graph
	{ 
		Graph g = new Graph(4); 

		g.addEdge(0, 1); 
		g.addEdge(0, 2); 
		g.addEdge(1, 2); 
		g.addEdge(2, 3); 
		g.BFS(2); 
	} 
} 
